src/aws-cpp-sdk-core/include/smithy/client/AwsSmithyClientBase.h

```diff
@@ -131,12 +139,17 @@ namespace client
                               Aws::Http::HttpMethod method,
                               EndpointUpdateCallback&& endpointCallback,
                               ResponseHandlerFunc&& responseHandler,
-                              std::shared_ptr<Aws::Utils::Threading::Executor> pExecutor) const;
+                              std::shared_ptr<Aws::Utils::Threading::Executor> pExecutor,
```

Comment:
the same here
I'd try to avoid leaking details to the base client class.
This API will be used by ~400 services, and having 2 extra arguments only for few of them is inconvenient in support.

Reply:
Same reasoning as above, my first attempt was decorator class which necessitated way more changes given the templated design. This was the way i could minimize it